### PR TITLE
feat(dropdownmenu): add props to support loading indicator

### PR DESCRIPTION
**Pull request by jfoo1984**
Allow loading and loadingText to be passed in to configure MUI Autocomplete.

## Summary

**Gene (Dropdown Menu)**
Shortcut ticket: (CZ ID ticket) https://app.shortcut.com/idseq/story/195075/nice-to-have-add-loading-screen-for-taxon-dropdown

Summary of ticket: Add loading indicator for search dropdown

## Checklist

- [x] Default Story in Storybook
- [x] LivePreview Story in Storybook
- [x] Test Story in Storybook
- [ ] Tests written
- [ ] Variables from `defaultTheme.ts` used wherever possible
- [ ] If updating an existing component, depreciate flag has been used where necessary
- [ ] Chromatic build verified by @chanzuckerberg/sds-design

### CODE REVIEW

**Review by masoudmanson — APPROVED** [review]
Awesome work 💯  Thank you Jerry 🥇🎉🙏🏻 @jfoo1984 
I believe we also need a designer's opinion. Connor might be able to assist us with the visual appearance of the loading indicator inside the menu. @clarsen-czi

File: src/core/Dropdown/index.stories.tsx [hunk 1]
```diff
@@ -65,6 +66,36 @@ MultipleSelectWithSearch.args = {
...
 
+export const MultipleSelectWithSearchNoOptionsText = Template.bind({});
+MultipleSelectWithSearchNoOptionsText.args = {
+  MenuSelectProps: {
+    noOptionsText: "No results",
+  },
+  label: LABEL,
+  multiple: true,
+  onChange: noop,
+  // options: [],
```
Comment (mmmmmmaya) [inline src/core/Dropdown/index.stories.tsx:77]: is this meant to be commented out? if so, should we just remove it?

**Comment by jfoo1984** [comment]
@masoudmanson That loading indicator is customizable though.  I just put it in there as an example in the story, but whatever is passed in as `loadingText` will get shown (it can be a string or node)

File: src/core/Dropdown/index.stories.tsx [hunk 2]
```diff
@@ -65,6 +66,36 @@ MultipleSelectWithSearch.args = {
...
+  search: true,
+};
```
Comment (jfoo1984) [inline src/core/Dropdown/index.stories.tsx:79]: This particular example / story was not added in this PR, but I thought it'd be useful to illustrate in Storybook.  Let me know if that's not something we want to add and I can remove it.

File: package.json [hunk 3]
```diff
@@ -30,7 +30,7 @@
     "@testing-library/react": "^12.1.2",
     "@types/jest": "^27.0.2",
     "@types/react": "^17.0.2",
-    "@types/react-dom": "^17.0.1",
+    "@types/react-dom": "^17.0.2",
```
Comment (jfoo1984) [inline package.json:33]: I was having issues with snapshot tests for the `Dropdown` component, an error about invalid hooks.  When I searched the error, possible cause were `react-dom` and `react` being different versions.  Aligning versions seemed to fix the issue, so I'm committing this.

File: src/core/Radio/index.tsx [hunk 4]
```diff
@@ -13,8 +13,9 @@ export interface RadioProps
  * @see https://v4.mui.com/components/radio-buttons/
  */
 const RadioButton = (props: RadioProps): JSX.Element => {
+  const { stage } = props;
```
Comment (jfoo1984) [inline src/core/Radio/index.tsx:16]: this was a linter warning that I fixed while trying to resolve other issues.  It's not strictly related, but I figured it doesn't hurt to get rid of a linter warning.

File: src/core/Checkbox/index.tsx [hunk 5]
```diff
@@ -14,8 +14,9 @@ export interface CheckboxProps
  * @see https://v4.mui.com/components/checkboxes/
  */
 const Checkbox = (props: CheckboxProps): JSX.Element => {
+  const { stage } = props;
   let newProps: MUICheckboxProps;
-  switch (props.stage) {
+  switch (stage) {
```
Comment (jfoo1984) [inline src/core/Checkbox/index.tsx:19]: this was a linter warning that I fixed while trying to resolve other issues. It's not strictly related, but I figured it doesn't hurt to get rid of a linter warning.

**Pull request merged**